src/kernels/MIOpenReduceCalculation.hpp

```diff
@@ -23,38 +23,56 @@
  * SOFTWARE.
  *
  *******************************************************************************/
-#ifndef GUARD_KERNELS_MIOPENREDUCECALCULATION_HPP
-#define GUARD_KERNELS_MIOPENREDUCECALCULATION_HPP
+#pragma once
 
 enum class ReduceCalculationOp_t
 {
-    Prod = 1,
+    First_ = 1,
+    Prod   = First_,
     Sum,
-    First_ = Prod,
-    Last_  = Sum,
+    lOR,  // Logical OR, to distinguish from bitwise OR
+    lAND, // Logical AND, to distinguish from bitwise AND
+    Last_ = lAND,
```

Comment:
why do we need `First_` and `Last_`

Reply:
This is used to validate whether an operation is within a defined range. For instance, to check if the input corresponds to a valid operation enum, I can use the following logic:
```
if (op_enum_input < ReduceCalculationOp_t::First_ || op_enum_input > ReduceCalculationOp_t::Last_) {
    return invalid;
}
```
By using `First_` and `Last_`, I can determine the total set of supported op_enum values.